instant-xml/src/lib.rs

```diff
@@ -10,15 +10,47 @@ pub use macros::{FromXml, ToXml};
 pub mod parse;
 
 pub trait ToXml {
-    fn write_xml<W: fmt::Write>(&self, write: &mut W) -> Result<(), Error>;
+    fn write_xml<W: fmt::Write>(
+        &self,
+        write: &mut W,
+        parent_prefixes: Option<&mut BTreeSet<&str>>,
```

Comment:
Let's wrap `prefixes` in an opaque `Serializer` type and make it required. `Serializer` should implement `Default` so that callers can just pass `Serializer::default()` (instead of `None`).

Reply:
I changed it a little bit, please take a look